Source/Validation.swift

```diff
@@ -34,9 +39,117 @@ extension Request {
         case failure(Error)
     }
 
-    /// A closure used to validate a request that takes a URL request and URL response, and returns whether the
+    fileprivate struct MIMEType {
```

Comment:
I'd vote for keeping it hidden until someone actually requests it. I haven't thought this myself yet.